src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1021,15 +1022,30 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
 
             // apply http configuration configuration
             configSection = (JObject)config["http"];
-            string routePrefix = ScriptConstants.DefaultHttpRoutePrefix;
+            var httpConfig = new HttpConfiguration
+            {
+                HttpRoutePrefix = ScriptConstants.DefaultHttpRoutePrefix
+            };
             if (configSection != null)
             {
                 if (configSection.TryGetValue("routePrefix", out value))
                 {
-                    routePrefix = (string)value;
+                    httpConfig.HttpRoutePrefix = (string)value;
```

Comment:
Now that you have a type to hold that configuration, could we simplify this (get rid of the if statements) just by deserializing the config?